wa/output_processors/postgresql.py

```diff
@@ -0,0 +1,761 @@
+            if augmentation in self.augmentations_already_added:
+                continue
+            self.current_augmentation_uuid = uuid.uuid4()
+            self.cursor.execute(self.command_create_augmentation,
+                                (
+                                    self.current_augmentation_uuid,
+                                    self.run_uuid,
+                                    augmentation
+                                )
+                                )
+            self.insert_parameters('augmentation',
+                                   output.run_config.augmentations[augmentation],
+                                   owner_id=self.current_augmentation_uuid,
+                                   job_specific=True)
+        self.conn.commit()
+
+    def insert_parameters(self, parameter_type, parameter_dict, owner_id=None, job_specific=False):
+        # Note, currently no augmentation parameters are workload specific, but in the future
+        # this may change
+        run_uuid = self.run_uuid
+        job_uuid = None  # Default initial value
+        augmentation_id = None
+        resource_getter_id = None
+        if parameter_type == "workload":
+            job_uuid = self.current_job_uuid
+        elif parameter_type == "resource_getter":
+            job_uuid = None
+            resource_getter_id = owner_id
+        elif parameter_type == "augmentation":
+            if job_specific:
+                job_uuid = self.current_job_uuid
+            augmentation_id = owner_id
+        elif parameter_type == "runtime":
+            job_uuid = self.current_job_uuid
+        else:
+            # boot parameters are not implemented
+            # device parameters are redundant due to the targets table
+            raise NotImplementedError
+        for parameter in parameter_dict:
+            self.current_parameter_uuid = uuid.uuid4()
+            from wa.utils.types import cpu_mask
+            if type(parameter_dict[parameter]) == cpu_mask:
+            self.cursor.execute(self.command_create_parameter,
+                                (
+                                    self.current_parameter_uuid,
+                                    run_uuid,
+                                    job_uuid,
+                                    augmentation_id,
+                                    resource_getter_id,
+                                    parameter,
+                                    str(parameter_dict[parameter]),
+                                    str(type(parameter_dict[parameter])),
+                                    parameter_type
+                                )
+                                )
+
+    def connect_to_database(self):
+        dsn = "dbname={} user={} password={} host={} port={}".format(
+            self.dbname, self.username, self.password, self.host, self.port)
+        try:
+            self.conn = connect(dsn=dsn)
+        except Exception as e:
```

Comment:
Do we know what sort of exceptions will be generated here rather than catching everything?